### PR TITLE
build_iterable_dataset_from_iterator

### PR DESCRIPTION
**Pull request by cpuhrsch**

### CODE REVIEW

File: examples/text_classification/iterable_train.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+from torchtext.vocab import build_vocab_from_iterator
+
+from tqdm import tqdm
+
+
+def generate_batch(batch):
+
+    def generate_offsets(data_batch):
+        offsets = [0]
+        for entry in data_batch:
```
Comment (mttk) [inline examples/text_classification/iterable_train.py]: This could be replaced by `np.cumsum`

File: examples/text_classification/iterable_train.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            offsets.append(offsets[-1] + len(entry))
+        offsets = torch.tensor(offsets[:-1])
+        return offsets
+
+    cls = torch.tensor([entry[0] for entry in batch])
+    text = [entry[1] for entry in batch]
+    offsets = generate_offsets(text)
+    text = torch.cat(text)
+    return text, offsets, cls
```
Comment (mttk) [inline examples/text_classification/iterable_train.py]: perhaps `label` instead of `cls`?

File: examples/text_classification/iterable_train.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        with torch.no_grad():
+            output = model(text, offsets)
+            accuracy = (output.argmax(1) == cls).float().mean().item()
+            total_accuracy.append(accuracy)
+    print("Test - Accuracy: {}".format(sum(total_accuracy) / len(total_accuracy)))
+
+def csv_iterator(data_path, ngrams, vocab, epochs=1, is_worker=False):
+    tokenizer = get_tokenizer("basic_english")
+    with io.open(data_path, encoding="utf8") as f:
+        f.seek(0, 2)
```
Comment (mttk) [inline examples/text_classification/iterable_train.py]: This part could be pretty confusing if not commented

File: examples/text_classification/iterable_train.py (hunk 1 above)
Comment (cpuhrsch) [reply examples/text_classification/iterable_train.py]: We don't want a hard numpy dependency, but I agree that'd be prettier

File: examples/text_classification/iterable_train.py (hunk 3 above)
Comment (cpuhrsch) [reply examples/text_classification/iterable_train.py]: Agreed, we'll work through all examples and add lots of comments.

File: examples/text_classification/iterable_train.py (hunk 2 above)
Comment (cpuhrsch) [reply examples/text_classification/iterable_train.py]: Sure
